packages/melos/lib/src/command_runner/init.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+
+import '../../melos.dart';
+import '../common/utils.dart';
+import 'base.dart';
+
+class InitCommand extends MelosCommand {
+  InitCommand(super.config) {
+    argParser.addOption(
+      'directory',
+      abbr: 'd',
+      help: 'Directory to create project in. Defaults to the workspace name.',
+    );
+
+    argParser.addMultiOption(
+      'packages',
+      abbr: 'p',
+      help: 'Comma separated glob paths to add to the melos workspace.',
+    );
+
+    argParser.addOption(
+      'project',
+      abbr: 'P',
+      help: 'Project name to be used in pubspec.yaml. '
+          'Defaults to the workspace name.',
+    );
+  }
+
+  @override
+  final String name = 'init';
+
+  @override
+  final String description = 'Initialize a new Melos workspace.';
+
+  @override
+  Future<void> run() {
+    final workspaceDefault = p.basename(Directory.current.absolute.path);
+    final workspaceName = argResults!.rest.firstOrNull ??
+        promptInput(
+          'Enter your workspace name',
+          defaultsTo: workspaceDefault,
+        );
+    final directory = argResults!['directory'] as String? ??
+        promptInput(
+          'Enter the directory',
+          defaultsTo: workspaceDefault != workspaceName ? workspaceName : '.',
+        );
+    final packages = argResults!['packages'] as List<String>?;
+    final project = argResults!['project'] as String? ??
+        promptInput(
+          'Enter the project name',
+          defaultsTo: workspaceName,
+        );
+    final useAppsDir = promptBool(
+      message: 'Do you want to add the apps directory to the list of packages?',
```

Comment:
```suggestion
      message: 'Do you want to add the apps directory?',
```
We are actually creating the `apps` directory right? I don't think we need to specify that it will be added in the melos.yaml file too, and the wording as is is slightly confusing.